hnn_core/cells_default.py

```diff
@@ -178,19 +180,22 @@ def _cell_L5Pyr(override_params, pos=(0., 0., 0), gid=0.):
                 partial(_exp_g_at_dist, zero_val=1e-6,
                         exp_term=3e-3, offset=0.0)
 
-    topology = [
-        # Distal (Apical)
-        ['soma', 1, 'apical_trunk', 0],
-        ['apical_trunk', 1, 'apical_1', 0],
-        ['apical_1', 1, 'apical_2', 0],
-        ['apical_2', 1, 'apical_tuft', 0],
-        # apical_oblique comes off distal end of apical_trunk
-        ['apical_trunk', 1, 'apical_oblique', 0],
-        # Proximal (basal)
-        ['soma', 0, 'basal_1', 0],
-        ['basal_1', 1, 'basal_2', 0],
-        ['basal_1', 1, 'basal_3', 0]
-    ]
+    cell_tree = {
+        ('soma', 0): [('soma', 1), ('basal_1', 0)],
+        ('soma', 1): [('apical_trunk', 0)],
+        ('apical_trunk', 0): [('apical_trunk', 1)],
+        ('apical_trunk', 1): [('apical_1', 0), ('apical_oblique', 0)],
+        ('apical_1', 0): [('apical_1', 1)],
+        ('apical_1', 1): [('apical_2', 0)],
+        ('apical_2', 0): [('apical_2', 1)],
+        ('apical_2', 1): [('apical_tuft', 0)],
+        ('apical_tuft', 0): [('apical_tuft', 1)],
+        ('apical_oblique', 0): [('apical_oblique', 1)],
+        ('basal_1', 0): [('basal_1', 1)],
```

Comment:
for sake of clarity, do you mind separating the connections where the section name is same with those that connect different sections? Something like:

```py
    cell_tree = {
        ('apical_1', 0): [('apical_1', 1)],
        ('basal_1', 0): [('basal_1', 1)],
        ('apical_2', 0): [('apical_2', 1)],
        ('apical_tuft', 0): [('apical_tuft', 1)],
        ('apical_oblique', 0): [('apical_oblique', 1)],
        # different sections connected
        ('soma', 0): [('soma', 1), ('basal_1', 0)],
        ('soma', 1): [('apical_trunk', 0)],
        ('apical_trunk', 0): [('apical_trunk', 1)],
        ('apical_trunk', 1): [('apical_1', 0), ('apical_oblique', 0)],
        ('apical_1', 1): [('apical_2', 0)],
        ('apical_2', 1): [('apical_tuft', 0)]
}
```

Reply:
Sure no problem